pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -895,14 +1012,198 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, podStatus *kubecontaine
+					return err
+				}
+			}
+			if len(podContainerChanges.ContainersToUpdate[rName]) > 0 {
+				if err = m.updatePodContainerResources(pod, podStatus, rName, podContainerChanges.ContainersToUpdate[rName]); err != nil {
+					klog.ErrorS(err, "updatePodContainerResources failed", "pod", format.Pod(pod), "resource", rName)
+					return err
+				}
+			}
+			if newPodCgValue < currPodCgValue {
+				err = setPodCgroupConfig(rName)
+			}
+			return err
+		}
+		if len(podContainerChanges.ContainersToUpdate[memLimit]) > 0 || podContainerChanges.UpdatePodResources {
+			currentPodMemoryLimit, err := pcm.GetPodCgroupMemoryConfig(pod)
+			if err != nil {
+				klog.ErrorS(err, "GetPodCgroupMemoryConfig failed", "pod", pod.Name)
+				return
+			}
+			currentPodMemoryUsage, err := pcm.GetPodCgroupMemoryUsage(pod)
+			if err != nil {
+				klog.ErrorS(err, "GetPodCgroupMemoryUsage failed", "pod", pod.Name)
+				return
+			}
+			if currentPodMemoryUsage >= uint64(*podResources.Memory) {
+				klog.ErrorS(nil, "Aborting attempt to set pod memory limit less than current memory usage", "pod", pod.Name)
+				return
+			}
+			if errResize := resizeContainers(memLimit, int64(currentPodMemoryLimit), *podResources.Memory); errResize != nil {
+				return
+			}
+		}
+		if len(podContainerChanges.ContainersToUpdate[cpuRequest]) > 0 || podContainerChanges.UpdatePodResources {
+			_, _, currentPodCPUShares, err := pcm.GetPodCgroupCpuConfig(pod)
+			if err != nil {
+				klog.ErrorS(err, "GetPodCgroupCpuConfig failed", "pod", pod.Name)
+				return
+			}
+			if errResize := resizeContainers(cpuRequest, int64(currentPodCPUShares), int64(*podResources.CpuShares)); errResize != nil {
+				return
+			}
+		}
+		if len(podContainerChanges.ContainersToUpdate[cpuLimit]) > 0 || podContainerChanges.UpdatePodResources {
+			currentPodCpuQuota, _, _, err := pcm.GetPodCgroupCpuConfig(pod)
+			if err != nil {
+				klog.ErrorS(err, "GetPodCgroupCpuConfig failed", "pod", pod.Name)
+				return
+			}
+			if errUpdate := resizeContainers(cpuLimit, currentPodCpuQuota, *podResources.CpuQuota); errUpdate != nil {
+				return
+			}
+		}
+	}
+
+	// Step 8: start containers in podContainerChanges.ContainersToStart.
 	for _, idx := range podContainerChanges.ContainersToStart {
 		start("container", containerStartSpec(&pod.Spec.Containers[idx]))
 	}
 
 	return
 }
 
+func (m *kubeGenericRuntimeManager) updatePodContainerResources(pod *v1.Pod, podStatus *kubecontainer.PodStatus, resourceName string, containersToUpdate []containerToUpdateInfo) error {
+	type updatedResourcesInfo struct {
+		cStatus *kubecontainer.Status
+		rLimits v1.ResourceList
+		rAllocs v1.ResourceList
+	}
+	updatedResourcesMap := make(map[kubecontainer.ContainerID]updatedResourcesInfo)
+
+	for _, cInfo := range containersToUpdate {
+		container := cInfo.apiContainer
+		containerStatus := cInfo.apiContainerStatus
+		specLimits := container.Resources.Limits
+		statusAllocs := containerStatus.ResourcesAllocated
```

Comment:
It is a bit hard to read, especially this `statusAllocs`.

It might be better to name them `desiredLimits`, `desiredRequests`, `currentLimits`, `curretnRequests`.

Reply:
done. this is much cleaner. i fixed the computePodResizeActions as well to do the same and it's a lot easier to read even for me.